zookeeper-api/src/main/java/org/apache/helix/zookeeper/impl/factory/SharedZkClientFactory.java

```diff
@@ -45,21 +45,12 @@ protected SharedZkClientFactory() {
   @Override
   public RealmAwareZkClient buildZkClient(
       RealmAwareZkClient.RealmAwareZkConnectionConfig connectionConfig,
-      RealmAwareZkClient.RealmAwareZkClientConfig clientConfig,
-      MetadataStoreRoutingData metadataStoreRoutingData) {
+      RealmAwareZkClient.RealmAwareZkClientConfig clientConfig) {
```

Comment:
This would be a conflict. Now my change is in, we need a rebase for this one.

Reply:
No worries.

I won't merge this in until other ZkClients have been merged in. I might include other ZkClients in the scope of this PR.